cupy/core/core.pyx

```diff
@@ -191,10 +192,12 @@ cdef class ndarray:
 
         if ver == 3:
             stream_ptr = stream_module.get_current_stream_ptr()
-            # TODO(leofang): check if we're using PTDS
             # CAI v3 says setting the stream field to 0 is disallowed
             if stream_ptr == 0:
-                stream_ptr = 1  # TODO(leofang): use runtime.streamLegacy
+                if _stream_module.is_ptds_enabled():
```

Comment:
Can we add a function to `cupy_backends.cuda.stream` that gives us the current default stream?
This if is being repeated a lot all over the code and I believe that this would simplify it.

Reply:
Sounds good. I call it `get_default_stream_ptr()` because I want to avoid using "current", which could be confusing if the default stream is not current.

Reply:
Done in 7a938a8.

Reply:
@emcastillo btw this function would never return `0` (the default null stream), only `1` or `2`.

Reply:
Didn't we roll back to `0` because of the NCCL bug with `cudaStreamLegacy`/`cudaStreamPerThread`?

Reply:
I think for now this function is used only to deal with CUDA Array Interface, which does not allow `0`. Should I add a comment to the function?

Reply:
I think I forgot about this with CUDA Array Interface, and probably unnecessary to add a comment. I think this is a very edgy case that is already fixed upstream anyway.